src/vaults/factories/AsyncVaultFactory.sol

```diff
@@ -49,4 +51,12 @@ contract AsyncVaultFactory is Auth, IVaultFactory {
         vault.deny(address(this));
         return vault;
     }
+
+    /// @notice Generate deterministic salt for CREATE2 deployment
+    /// @dev Uses keccak256 hash of encoded poolId, scId, and asset address to ensure
+    ///      deterministic vault addresses across all chains. Same inputs will always
+    ///      produce the same salt, enabling predictable cross-chain vault addresses.
+    function _generateSalt(PoolId poolId, ShareClassId scId, address asset) internal pure returns (bytes32 salt) {
+        salt = keccak256(abi.encode(poolId, scId, asset));
+    }
```

Comment:
Do we really need the function? Just to reduce the number of lines of the fix for the audits